credential/concept/template.go

```diff
@@ -0,0 +1,319 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package concept
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/nuts-foundation/go-did"
+	"github.com/thedevsaddam/gojsonq/v2"
+)
+
+// TypeField defines the concept/VC JSON path to a VC type
+const TypeField = "type"
+
+// IDField defines the concept/VC JSON path to a VC ID
+const IDField = "id"
+
+// IssuerField defines the concept/VC JSON path to a VC issuer
+const IssuerField = "issuer"
+
+// SubjectField defines the concept JSON path to a VC subject
+const SubjectField = "subject"
+
+// Template represents a json mapping template. In the template concept to json path mappings are stored.
+// Indices that need to be created by a DB are also created by the template.
+type Template interface {
+	// Indices returns a list of compound indices. Each entry contains the concept json path.
+	Indices() [][]string
+	// VCType returns the VC type.
+	VCType() string
```

Comment:
Do we need to bind 'concepts' to VCs in code, or can we let it be generic ('Object'/'Entity'?)? Then we could use it for our DID store as well?

Reply:
Aren't DID documents more static than VCs?